contracts/DrawCalculator.sol

```diff
@@ -397,16 +403,27 @@ contract DrawCalculator is IDrawCalculator, Ownable {
         pure
         returns (uint256[] memory)
     {
-        uint256[] memory masks = new uint256[](_prizeDistribution.matchCardinality);
+        
+        // calculate the non-zero length of the prize tiers
+        uint8 nonZeroPrizeTiersLength = 0;
+        for(uint i = 0; i < _prizeDistribution.tiers.length; i++) {
+            if(_prizeDistribution.tiers[i] > 0) {
+                nonZeroPrizeTiersLength++;
+            }
+        }
+        uint8 _matchCardinality = _prizeDistribution.matchCardinality;
+        uint8 createMasksLength = nonZeroPrizeTiersLength > _matchCardinality ? nonZeroPrizeTiersLength : _matchCardinality;
```

Comment:
seems to me this should always be the **smaller** of the two, not the larger.

If cardinality < number of tiers, then we'll never get to the higher tiers anyway.

if tiers < cardinality, then we don't need to hold more masks than the number of tiers.

Reply:
> if tiers < cardinality, then we don't need to hold more masks than the number of tiers.
Then we still need to bound the matching loop with masks.length as per your other comment